runtime/src/main/java/org/corfudb/protocols/wireprotocol/failuredetector/NodeRank.java

```diff
@@ -48,8 +50,34 @@ public boolean is(String endpoint) {
         return this.endpoint.equals(endpoint);
     }
 
-    public NodeRank(ByteBuf buf){
+    public NodeRank(ByteBuf buf) {
         endpoint = CorfuProtocolCommon.fromBuffer(buf, String.class);
         numConnections = CorfuProtocolCommon.fromBuffer(buf, Integer.class);
     }
+
+    @AllArgsConstructor
+    @EqualsAndHashCode
+    @Getter
+    @ToString
+    public static class NodeRankByResourceQuota implements Comparable<NodeRankByResourceQuota> {
```

Comment:
I don't want to add more data into NodeRank, it will help to keep NodeRank simple. All different types of ranks in this class have their own `comparator` strategy. If I add more into NodeRank I will need to extract comparators and use NodeRank everywhere and some part of that data will not be relevant to the required action. 
So, I believe keeping  them separate is a better strategy